lib/entropy/server/interfaces/main.py

```diff
@@ -6213,32 +6214,38 @@ def _generate_extra_download(path, down_type):
 
         myserver_repos = list(srv_set['repositories'].keys())
 
-        ### since we are handling more repositories, we need to make sure
-        ### that there are no packages in other repositories with same atom
-        ### and greater revision
-        rev_test_atom = mydata['atom']
-        max_rev = -1
-        for myrepo in myserver_repos:
-
-            # not myself
-            if myrepo == repository_id:
-                continue
-
-            mydbconn = self.open_server_repository(myrepo, read_only = True,
-                no_upload = True)
-
-            myrepo_package_ids = mydbconn.getPackageIds(rev_test_atom)
-            for myrepo_package_id in myrepo_package_ids:
-                myrev = mydbconn.retrieveRevision(myrepo_package_id)
-                if myrev > max_rev:
-                    max_rev = myrev
-
-        if max_rev >= revision:
-            max_rev += 1
-            revision = max_rev
+        if reset_revision:
+            revision = 0
             mydata['revision'] = revision
```

Comment:
These three lines are shared in if/else. Deduplicate them.

Reply:
take a look now if could be fine. thanks